app/src/main/java/org/openobservatory/ooniprobe/common/service/ServiceUtil.java

```diff
@@ -63,49 +60,37 @@ public static void stopJob(Context context) {
     }
 
     public static void callCheckInAPI(Application app) {
+        app.getServiceComponent().inject(d);
+
         BatteryManager batteryManager = (BatteryManager) app.getSystemService(Context.BATTERY_SERVICE);
-        PreferenceManager pm = app.getPreferenceManager();
-        //Double checks for charging and wifi
-        if (pm.testWifiOnly() &&
-                !ReachabilityManager.getNetworkType(app).equals(ReachabilityManager.WIFI))
-            return;
-        if (pm.testChargingOnly() &&
-                !batteryManager.isCharging())
-            return;
-        try {
-            OONISession session = EngineProvider.get().newSession(
-                    EngineProvider.get().getDefaultSessionConfig(
-                            app, BuildConfig.SOFTWARE_NAME, BuildConfig.VERSION_NAME, new LoggerArray())
-            );
-            OONIContext ooniContext = session.newContextWithTimeout(30);
-            session.maybeUpdateResources(ooniContext);
-			OONICheckInConfig config = new OONICheckInConfig(
-					BuildConfig.SOFTWARE_NAME,
-					BuildConfig.VERSION_NAME,
-                    ReachabilityManager.isOnWifi(app),
-                    ReachabilityManager.isCharging(app),
-					app.getPreferenceManager().getEnabledCategoryArr().toArray(new String[0]));
-			OONICheckInResults results = session.checkIn(ooniContext, config);
-            if (results.webConnectivity != null) {
-                ArrayList<String> inputs = new ArrayList<>();
-                for (OONIURLInfo url : results.webConnectivity.urls){
-                    inputs.add(url.url);
-                }
-                AbstractSuite suite = AbstractSuite.getSuite(app, "web_connectivity",
-                        inputs,"autorun");
-                if (suite != null) {
-                    app.getPreferenceManager().updateAutorunDate();
-                    app.getPreferenceManager().incrementAutorun();
-                    Intent serviceIntent = new Intent(app, RunTestService.class);
-                    serviceIntent.putExtra("testSuites", suite.asArray());
-                    serviceIntent.putExtra("storeDB", false);
-                    ContextCompat.startForegroundService(app, serviceIntent);
-                }
-            }
-        }
-        catch (Exception e) {
-            e.printStackTrace();
-            ThirdPartyServices.logException(e);
+        Boolean workingOnWifi = ReachabilityManager.getNetworkType(app).equals(ReachabilityManager.WIFI);
+        String[] categories = d.preferenceManager.getEnabledCategoryArr().toArray(new String[0]);
+        Boolean phoneCharging = batteryManager.isCharging();
+
+        OONICheckInConfig config = new OONICheckInConfig(
+                BuildConfig.SOFTWARE_NAME,
+                BuildConfig.VERSION_NAME,
+                workingOnWifi,
+                phoneCharging,
+                categories
+               );
```

Comment:
You left this outside the domain class because you couldn't created it?

Reply:
yes

Reply:
it will attempt to create in the constructor CheckInConfigWebConnectivity, and all class's that "import go.Seq;" or use implements Seq.Proxy can't be inited in testing or mocked.